Flashcards.ngalantino/Flashcards.ngalantino/DatabaseManager.cs

```diff
@@ -0,0 +1,270 @@
+
+    }
+
+    public void NewStack(String name)
+    {
+        using (SqlConnection connection = new SqlConnection(System.Configuration.ConfigurationManager.AppSettings.Get("dbConnection")))
+        {
+            connection.Open();
+
+            string sql = $"INSERT INTO flashcardstest.dbo.stacks (stack) VALUES ('{name}')";
+
+            SqlCommand command = new SqlCommand(sql, connection);
+
+            command.ExecuteNonQuery();
+        }
+    }
+
+    public void DeleteStack(String stack)
+    {
+        using (SqlConnection connection = new SqlConnection(System.Configuration.ConfigurationManager.AppSettings.Get("dbConnection")))
+        {
+            connection.Open();
+
+            string sql = $"DELETE FROM flashcardstest.dbo.stacks WHERE stack='{stack}'";
+
+            SqlCommand command = new SqlCommand(sql, connection);
+
+            command.ExecuteNonQuery();
+        }
+    }
+
+    public void AddFlashcard(Flashcard flashcard)
+    {
+        using (SqlConnection connection = new SqlConnection(System.Configuration.ConfigurationManager.AppSettings.Get("dbConnection")))
+        {
+
+            connection.Open();
+
+            string sql = $"INSERT INTO flashcardstest.dbo.flashcards (stack, front, back) VALUES('{flashcard.Stack}', '{flashcard.Front}', '{flashcard.Back}')";
+
+            SqlCommand command = new SqlCommand(sql, connection);
+
+            command.ExecuteNonQuery();
+
+        }
+    }
+
+    public void EditFlashcard(Flashcard flashcard)
+    {
+        using (SqlConnection connection = new SqlConnection(System.Configuration.ConfigurationManager.AppSettings.Get("dbConnection")))
+        {
+
+            connection.Open();
+
+            string sql = @$"UPDATE flashcardstest.dbo.flashcards
+                            SET front = '{flashcard.Front}', back = '{flashcard.Back}'
+                            WHERE id = '{flashcard.Id}'";
+
+            SqlCommand command = new SqlCommand(sql, connection);
+
+            command.ExecuteNonQuery();
+
+        }
+    }
+
+    public void DeleteFlashcard(Flashcard flashcard)
+    {
+        using (SqlConnection connection = new SqlConnection(System.Configuration.ConfigurationManager.AppSettings.Get("dbConnection")))
+        {
+            connection.Open();
+
+            string sql = $"DELETE FROM flashcardstest.dbo.flashcards WHERE id='{flashcard.Id}'";
+
+            SqlCommand command = new SqlCommand(sql, connection);
+
+            command.ExecuteNonQuery();
+        }
+    }
+
+    public void AddStudySession(String stack, DateTime date, int score)
+    {
+
+        using (SqlConnection connection = new SqlConnection(System.Configuration.ConfigurationManager.AppSettings.Get("dbConnection")))
+        {
+
+            connection.Open();
+
+            string sql = $"INSERT INTO flashcardstest.dbo.studysessions (stack, date, score) VALUES ('{stack}', '{date}', '{score}')";
+
+            SqlCommand command = new SqlCommand(sql, connection);
```

Comment:
You have an error in this SQL statement. Use parameters to avoid further errors. If in doubt, refer to the Dapper documentation.